chatter/src/contexts/LatestMessages/LatestMessages.js

```diff
@@ -1,19 +1,25 @@
-import React, { useState, createContext, useCallback } from 'react';
+import React, { createContext, useCallback, useRef } from 'react';
 import initialMessages from './constants/initialMessages';
+import { BehaviorSubject } from 'rxjs';
 
 const LatestMessagesContext = createContext({});
 
 export default LatestMessagesContext;
 
 export function LatestMessages({ children }) {
-  const [messages, setMessages] = useState(initialMessages);
+  // Convert useState to useRef+Observable is to avoid unnecessary child components rerendering when context changes.
+  const messagesRef = useRef(new BehaviorSubject(initialMessages));
 
   const setLatestMessage = useCallback((userId, value) => {
-    setMessages({ ...messages, [userId]: value });
-  }, [messages]);
+    const subject = messagesRef.current;
+    const messages = subject.getValue();
+
+    // Publish global latest messages change
+    subject.next({ ...messages, [userId]: value });
+  }, []);
```

Comment:
Using `useRef & Observable` instead of previous `useState` can avoid unnecessary child component re-rendering. Furthermore, avoid unnecessary Socket listener registers/deletions.